src/AutorestSwift/View Models/OperationViewModel.swift

```diff
@@ -53,37 +82,14 @@ struct OperationViewModel {
         self.name = operationName(for: operation.name)
         self.comment = ViewModelComment(from: operation.description)
 
-        var signatureComments: [String] = []
-        for param in operation.signatureParameters ?? [] where param.description != "" {
-            signatureComments.append("   - \(param.name) : \(param.description)")
-        }
-        self.signatureComment = ViewModelComment(from: signatureComments.joined(separator: "\n"))
-
-        var requiredQueryParams = [KeyValueViewModel]()
-        var requiredHeaders = [KeyValueViewModel]()
-        var optionalQueryParams = [KeyValueViewModel]()
-        var optionalHeaders = [KeyValueViewModel]()
-        var uriParams = [KeyValueViewModel]()
+        var params = OperationParameters()
         var pipelineContext = [KeyValueViewModel]()
 
-        for param in operation.parameters ?? [] {
-            guard let httpParam = param.protocol.http as? HttpParameter else { continue }
-
-            let viewModel = KeyValueViewModel(from: param, with: operation)
-
-            switch httpParam.in {
-            case .query:
-                viewModel.optional ? optionalQueryParams.append(viewModel) : requiredQueryParams.append(viewModel)
-            case .header:
-                viewModel.optional ? optionalHeaders.append(viewModel) : requiredHeaders
-                    .append(viewModel)
-            case .path,
-                 .uri:
-                uriParams.append(viewModel)
-            default:
-                continue
-            }
-        }
+        params = buildQueryParamsHeaders(
```

Comment:
In this case, since params is initially empty, you can just use
```Swift
var params = buildQueryParamsHeaders(
  parameters: operation.parameters,
  operation: operations
)
```
Or, even better, just move the implementation of `buildQueryParamsHeaders` into an init for `OperationParameters()`.

```Swift
struct OperationParamters() {
  init(parameters: [Parameters], operation: Operation) {
    // move all that logic in here
  }
}
```

Reply:
There are 2 cases: 
1.  when all header and query are empty.
2. when header, query are populated and i want to add more header/query to existing entries in those arrays

If i move all the buildQueryParamsHeaders() logic to init(), I need to copy that to the 2nd init ` init(operationParameters: OperationParameters) {`, 

So that will result in duplicate code.

Reply:
buildQueryParamsHeaders is called in 2 places in the init() of OperationViewModel

Reply:
I working on making this more concise.